pkg/microservice/aslan/core/service/service/service.go

```diff
@@ -249,6 +255,117 @@ func GetServiceOption(args *commonmodels.Service, log *zap.SugaredLogger) (*Serv
 	return serviceOption, nil
 }
 
+func CreateK8sWorkLoads(ctx context.Context, requestID, username string, productName string, workLoads []models.Workload, clusterID, namespace string, envName string, log *zap.SugaredLogger) error {
+	kubeClient, err := kube.GetKubeClient(clusterID)
+	if err != nil {
+		log.Errorf("[%s] error: %v", namespace, err)
+		return err
+	}
+	// 检查环境是否存在，envName和productName唯一
+	opt := &commonrepo.ProductFindOptions{Name: productName, EnvName: envName}
+	if _, err := commonrepo.NewProductColl().Find(opt); err == nil {
+		log.Errorf("[%s][P:%s] duplicate envName in the same product", envName, productName)
+		return e.ErrCreateEnv.AddDesc(e.DuplicateEnvErrMsg)
+	}
+
+	// todo Add data filter
+	var (
+		workloadsTmp []models.Workload
+		wg           sync.WaitGroup
+		mu           sync.Mutex
+	)
+
+	for _, workload := range workLoads {
+		wg.Add(1)
+
+		go func(tempWorkload models.Workload) {
+			defer wg.Done()
+			var bs []byte
+			switch tempWorkload.Type {
+			case setting.Deployment:
+				bs, _, err = getter.GetDeploymentYaml(namespace, tempWorkload.Name, kubeClient)
+			case setting.StatefulSet:
+				bs, _, err = getter.GetStatefulSetYaml(namespace, tempWorkload.Name, kubeClient)
+			}
+
+			if len(bs) == 0 || err != nil {
+				log.Errorf("not found yaml %v", err)
+				return
+			}
+			workloadsTmp = append(workloadsTmp, models.Workload{
+				EnvName:     envName,
+				Name:        tempWorkload.Name,
+				Type:        tempWorkload.Type,
+				ProductName: productName,
+			})
+			mu.Lock()
+			defer mu.Unlock()
+			if _, err = CreateServiceTemplate(username, &models.Service{
+				ServiceName:  tempWorkload.Name,
+				Yaml:         string(bs),
```

Comment:
you can get the containers directly, don't parse it from yaml

Reply:
only for record ,no use